File: _DEMO/agent-server/tools/get_customer_summary/handler.js
// [CHG-001] Extracted from lib/mock-data.js.
// Delegates to get_open_items and get_customer_details to avoid duplicating fixtures.

import { execute as getOpenItems }      from '../get_open_items/handler.js';
import { execute as getCustomerDetails } from '../get_customer_details/handler.js';

export async function execute(args, context) {
  const [openItemsRes, customerRes] = await Promise.all([
    getOpenItems({}, context),
    getCustomerDetails({}, context),
  ]);
  const items      = openItemsRes.d.results;
  const totalOpen  = items.reduce((sum, r) => sum + parseFloat(r.OpenAmountInTransCrcy), 0);
  return {
    customer:        customerRes.d,
    openItems:       items,
    totalOpenAmount: totalOpen.toFixed(2),
    currency:        'USD',
    overdueCount:    items.filter((r) => r.IsOverdue).length,
  };
}

File: _DEMO/agent-server-py/tools/get_customer_summary/handler.py
# [C03-F12] Python equivalent of tools/get_customer_summary/handler.js
# Delegates to get_open_items and get_customer_details — no data duplication.

from tools.get_open_items.handler import execute as get_open_items
from tools.get_customer_details.handler import execute as get_customer_details


async def execute(args: dict, context: dict) -> dict:
    import asyncio
    open_items_res, customer_res = await asyncio.gather(
        get_open_items({}, context),
        get_customer_details({}, context),
    )
    items = open_items_res["d"]["results"]
    total_open = sum(float(r["OpenAmountInTransCrcy"]) for r in items)
    return {
        "customer": customer_res["d"],
        "openItems": items,
        "totalOpenAmount": f"{total_open:.2f}",
        "currency": "USD",
        "overdueCount": sum(1 for r in items if r["IsOverdue"]),
    }
